doc/developer_guide/psyir_backends.rst

```diff
@@ -241,7 +241,37 @@ following simplified way (as if it were a function)::
 The primary reason for providing the above (functor) interface is to
 hide users from the use of the visitor pattern. This is the interface
 to expose to users (which is why `_visit` is used for the visitor
-method, rather than `visit`).
+method, rather than `visit`). An important characteristic of the `__call__`
+method is that it will manage the lowering of DSL-concepts because the
+backends should not provide specific visitors for concepts that do not relate
+directly to the language domain (more information about the lowering step is
+provided in the :ref:`psy_layer_backends` section below). This step is done
+internally without exposing side effects (e.g. modifications to the provided
+tree). This is important because it allows to turn on or off the generation
+of backend code without altering the result, which is often convenient for
+debugging and development. For instance the walk statement in the following
+example will return the same nodes regardless of whether the print statement
```

Comment:
"...nodes, regardless of whether or not the ....commented out::"

Reply:
Done